git_machete/utils.py

```diff
@@ -168,7 +168,12 @@ def run_cmd(cmd: str, *args: str, **kwargs: Any) -> int:
     elif verbose_mode:
         sys.stderr.write(flat_cmd + "\n")
 
-    exit_code: int = subprocess.call([cmd] + list(args), **kwargs)
+    completed_process: subprocess.CompletedProcess[bytes] = subprocess.run([cmd] + list(args), stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs)
+    exit_code: int = completed_process.returncode
+
+    if exit_code != 0:
+        sys.stdout.writelines(completed_process.stdout.decode('utf-8'))
```

Comment:
Hmmm this is a change in logic now... since AFAICS that the entire output will only be printed **once the process is finished**, and not while the process is running. I can't think now of any specific command that will be affected (maybe `git rebase`?) but I doubt this will NOT have any consequence at all

Reply:
Oh okay, I wish I've seen the read PR description more thoroughly first 😅

Reply:
Hmmm how about mocking `run_cmd` or `subprocess.call` in the tests? Don't proceed however if it requires too much hacking/"engineering", we can live with #207 unresolved (#228 would be still nice to fix, tho)

Reply:
> So, main question is to capture all stdout/err or just mock it in tests?

Generally,  **no change to logic in production code** should happen, only to test code